waku/node/v2/waku_types.nim

```diff
@@ -32,11 +32,29 @@ type
 
   QueryHandlerFunc* = proc(response: HistoryResponse) {.gcsafe, closure.}
 
+
+  Index* = object
+    ## This type contains the  description of an index used in the pagination of waku messages
+    digest*: string
+    timestamp*: string 
+
+  IndexedWakuMessage* = object 
+    msg*: WakuMessage
+    index*: Index
+
+  PagingInfo* = object
+    ## This type holds the information needed for the pagination 
+    pageSize*: int
+    cursor*: Index
+    direction*: bool
+
   HistoryQuery* = object
     topics*: seq[string]
+    pagingInfo*: PagingInfo
```

Comment:
Might want to have this be optional for implementations that don't want to care about this.

Reply:
Sure, for a HistoryQuery with an unspecified PagingInfo, no pagination will take place.